go/vt/srvtopo/query.go

```diff
@@ -173,9 +169,9 @@ func (q *resilientQuery) getCurrentValue(ctx context.Context, wkey fmt.Stringer,
 	}
 	entry.mutex.Lock()
 
-	if entry.value != nil {
-		return entry.value, nil
+	if entry.lastError != nil {
+		return nil, entry.lastError
 	}
 
-	return nil, entry.lastError
+	return entry.value, nil
 }
```

Comment:
So,  If we have reached here and still there is entry with `lastError` which means the last refresh have failed, we do not want to return the cached value but instead return the error.